core/ecschema-metadata/src/test/Metadata/SchemaFromJson.test.ts

```diff
@@ -0,0 +1,334 @@
+        ...schemaAJson,
+        references: [
+          {
+            name: "DummyReference",
+            version: "01.00.01",
+          },
+        ],
+        customAttributes: [
+          {
+            className: "DummyReference.customAttributeOne",
+            showClasses: true,
+          },
+        ],
+      };
+
+      expect(await Schema.fromJson(schemaA, context)).to.not.be.undefined;
+    });
+
+    it("should throw an error and not allow the creation of the schema with top level custom attribute and no reference defined", async () => {
+      const _dummyRefOne = await Schema.fromJson({
+        ...dummyRefJson,
+        items: {
+          customAttributeOne: {
+            schemaItemType: "CustomAttributeClass",
+            appliesTo: "AnyClass",
+          },
+        },
+      }, context);
+
+      const schemaA = {
+        ...schemaAJson,
+        customAttributes: [
+          {
+            className: "DummyReference.customAttributeOne",
+            showClasses: true,
+          },
+        ],
+        items: {
+          testClassOne: {
+            schemaItemType: "EntityClass",
+            description: "Test class one",
+          },
+          testClassTwo: {
+            schemaItemType: "EntityClass",
+            description: "Test class two",
+          },
+        },
+      };
+
+      await expect(Schema.fromJson(schemaA, context)).to.be.rejectedWith(ECObjectsError, "Unable to load custom attribute DummyReference.customAttributeOne from container SchemaA, DummyReference reference not defined");
+    });
+
+    it("should throw an error and not allow the creation of the schema with top level custom attribute and its reference not defined", async () => {
+      const _dummyRefOne = await Schema.fromJson({
+        ...dummyRefJson,
+        items: {
+          customAttributeOne: {
+            schemaItemType: "CustomAttributeClass",
+            appliesTo: "AnyClass",
+          },
+        },
+      }, context);
+
+      const _dummyRefTwo = await Schema.fromJson({
+        ...dummyRefJsonTwo,
+        items: {
+          customAttributeOneRefTwo: {
+            schemaItemType: "CustomAttributeClass",
+            appliesTo: "AnyClass",
+          },
+        },
+      }, context);
+
+      const schemaA = {
+        ...schemaAJson,
+        references: [
+          {
+            name: "DummyReference",
+            version: "01.00.01",
+          },
+        ],
+        customAttributes: [
+          {
+            className: "DummyReference.customAttributeOne",
+            showClasses: true,
+          },
+          {
+            className: "DummyReferenceTwo.customAttributeOneRefTwo",
+            showClasses: true,
+          },
+        ],
+      };
+
+      await expect(Schema.fromJson(schemaA, context)).to.be.rejectedWith(ECObjectsError, "Unable to load custom attribute DummyReferenceTwo.customAttributeOneRefTwo from container SchemaA, DummyReferenceTwo reference not defined");
+    });
```

Comment:
You don't need both of these tests if you are using the getReferenceSync method because that method handles accessing references directly and should already have tests to ensure it works with no references

Reply:
Removed.